pkg/reconciler/v1alpha1/route/route.go

```diff
@@ -206,21 +206,44 @@ func (c *Reconciler) Reconcile(ctx context.Context, key string) error {
 	return err
 }
 
-func (c *Reconciler) reconcile(ctx context.Context, route *v1alpha1.Route) error {
+func (c *Reconciler) reconcile(ctx context.Context, r *v1alpha1.Route) error {
 	logger := logging.FromContext(ctx)
-	route.Status.InitializeConditions()
+	r.Status.InitializeConditions()
 
-	logger.Infof("Reconciling route :%v", route)
-	logger.Info("Creating/Updating placeholder k8s services")
-	if err := c.reconcilePlaceholderService(ctx, route); err != nil {
+	logger.Infof("Reconciling route :%v", r)
+	// configure traffic based on the RouteSpec.
+	traffic, err := c.configureTraffic(ctx, r)
+	if traffic == nil || err != nil {
+		// Traffic targets aren't ready, no need to configure child resources.
 		return err
 	}
 
-	// Call configureTrafficAndUpdateRouteStatus, which also updates the Route.Status
-	// to contain the domain we will use for Gateway creation.
-	if _, err := c.configureTraffic(ctx, route); err != nil {
+	logger.Info("Updating targeted revisions.")
+	// In all cases we will add annotations to the referred targets.  This is so that when they become
+	// routable we can know (through a listener) and attempt traffic configuration again.
+	if err := c.reconcileTargetRevisions(ctx, traffic, r); err != nil {
+		logger.Errorf("Failed to update target revisions: %v", err)
```

Comment:
this no longer returns an error?

Reply:
I did not touch the part in this change. so I guess it never returns error.
ping @greghaynes for confirm.

Reply:
Correct - this never returned an error. This might be a newbie mistake but my thinking was this isn't a hard failure and all the following logic does not depend on the annotation being properly set. Thinking a bit more though this is probably not the correct behavior as it will prevent us from retrying on this resource?

Reply:
https://github.com/knative/serving/pull/2333

Reply:
@greghaynes Thanks for your reply! I intended to update here to return error, in order not to block this PR, but the unit test is failing (not only the one you modified in #2333, but the case `TestCreateRouteForOneReserveRevision` in route_test.go also fail with error `TestCreateRouteForOneReserveRevision.route-controller    route/reconcile_resources.go:216    Unable to set revision annotation: invalid JSON document    {"knative.dev/controller": "route-controller"}`)

@mattmoor Can we just ignore this issue here, as it will be covered in #2333 for sure.

Reply:
The fix for that failure just merged: https://github.com/knative/pkg/pull/145

I agree with ignoring that issue for now and well pull in new pkg and the fix with #2333